### ISSUE TITLE
BUG: TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by unsignedrant** [numpy/numpy#21519]
### Describe the issue:

Use [sample.zip](https://github.com/numpy/numpy/files/8701836/sample.zip) to run the example

I need to know whether this is a numpy issue or a cython issue or pycharm's. The error only happens in pycharm debugger under ambiguous circumstances, and it is a known [issue](https://youtrack.jetbrains.com/issue/PY-52137). There are [mentions](https://youtrack.jetbrains.com/issue/PY-52137/Debugger-fails-with-numpy-getlimits-TypeError-NoneType-object-is#focus=Comments-27-6068544.0-0) that it's a cython [issue](https://github.com/cython/cython/pull/4735) despite the failure occurring in `numpy/core/getlimits.py Line 649`. I need to know where the problem originates and if it's a numpy issue, is there some merged fix?

**Update:** The error is gone after building cython with the fix, numpy, pandas, and scikit-learn

### Reproduce the code example:

```python
import pandas as pd
from sklearn.feature_extraction.text import CountVectorizer

df = pd.read_parquet('sample.parquet')
corpus = df[df['cell_type'] == 'markdown']['source']
vec = CountVectorizer(ngram_range=(1, 1))
vec.fit(corpus)
```

### Error message:

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/numpy/core/getlimits.py", line 649, in __init__
    self.dtype = numeric.dtype(int_type)
TypeError: 'NoneType' object is not callable
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1491, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/user/Library/Application Support/JetBrains/PyCharm2022.1/scratches/scratch_3.py", line 7, in <module>
    vec.fit(corpus)
  File "/usr/local/lib/python3.10/site-packages/sklearn/feature_extraction/text.py", line 1283, in fit
    self.fit_transform(raw_documents)
  File "/usr/local/lib/python3.10/site-packages/sklearn/feature_extraction/text.py", line 1330, in fit_transform
    vocabulary, X = self._count_vocab(raw_documents, self.fixed_vocabulary_)
  File "/usr/local/lib/python3.10/site-packages/sklearn/feature_extraction/text.py", line 1224, in _count_vocab
    if indptr[-1] > np.iinfo(np.int32).max:  # = 2**31 - 1
  File "/usr/local/lib/python3.10/site-packages/numpy/core/getlimits.py", line 651, in __init__
    self.dtype = numeric.dtype(type(int_type))
TypeError: 'NoneType' object is not callable
python-BaseException
```

### NumPy/Python version information:

```
python: 3.10.4
numpy: 1.22.3
```

**Comment by unsignedrant** [comment]
@rkern I just rebuilt cython with the fix but now I'm facing another issue when I try to build numpy:

    Processing ./numpy
      Installing build dependencies ... done
      Getting requirements to build wheel ... error
      error: subprocess-exited-with-error
      
      × Getting requirements to build wheel did not run successfully.
      │ exit code: 1
      ╰─> [20 lines of output]
          Error in sitecustomize; set PYTHONVERBOSE for traceback:
          AssertionError:
          Traceback (most recent call last):
            File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
              main()
            File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
              json_out['return_val'] = hook(**hook_input['kwargs'])
            File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
              return hook(config_settings)
            File "/usr/local/Cellar/python@3.10/3.10.4/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
              return self._get_build_requires(
            File "/usr/local/Cellar/python@3.10/3.10.4/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
              self.run_setup()
            File "/usr/local/Cellar/python@3.10/3.10.4/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/setuptools/build_meta.py", line 281, in run_setup
              super(_BuildMetaLegacyBackend,
            File "/usr/local/Cellar/python@3.10/3.10.4/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
              exec(compile(code, __file__, 'exec'), locals())
            File "setup.py", line 64, in <module>
              raise RuntimeError(f'Cannot parse version {FULLVERSION}')
          RuntimeError: Cannot parse version 0+untagged.1.gc5a9996
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
    
    note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by unsignedrant** [comment]
**Update:** It's not a cython issue, I just installed the cython [branch](https://github.com/cython/cython/tree/0.29.x) having the assumed [fix](https://github.com/cython/cython/pull/4735) followed by building numpy and was able to reproduce the same problem in pycharm debugger.

**Comment by seberg** [comment]
No, it almost certain is, but it is *not* triggered by NumPy, you have to rebuild scikit-learn, or pandas, or whatever type of packages you are using that use cython.
NumPy actually only uses cython for its random numbers.

**Comment by unsignedrant** [comment]
I can confirm, the error is gone after building scikit-learn, pandas

**Comment by Mark-BC** [comment]
I had this same problem:  Using PyCharm, but **only if running In debug mode**, executing

> `my_dtype = numpy.core.numeric.dtype(numpy.int32)`

would cause

> TypeError: 'NoneType' object is not callable

I solved it by upgrading Pandas from 1.4.2 to 1.4.3
